components/emoji_picker/emoji_picker.jsx

```diff
@@ -404,12 +396,96 @@ export default class EmojiPicker extends React.Component {
                     className='emoji-picker__search'
                     type='text'
                     onChange={this.handleFilterChange}
+                    onKeyDown={this.handleKeyDown}
                     placeholder={Utils.localizeMessage('emoji_picker.search', 'search')}
                 />
             </div>
         );
     }
 
+    emojiCurrentResults() {
+        const {cursor, filter} = this.state;
+        const categories = filter ? [CATEGORY_SEARCH_RESULTS] : Object.keys(this.state.categories);
+        let numEmojisLoaded = -1;
+        return (
+            <div
+                ref={(emojiPickerContainer) => {
```

Comment:
Also, there's a few more places that we're passing in a new function as the ref prop each time. These should all be methods of the EmojiPicker or just use string refs

Reply:
fixed all of these